src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -0,0 +1,161 @@
+import { CTX_INLINE_CHAT_VISIBLE, IInlineChatService } from 'vs/workbench/contrib/inlineChat/common/inlineChat';
+import { RunOnceScheduler } from 'vs/base/common/async';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { Iterable } from 'vs/base/common/iterator';
+
+const gutterInlineChatIcon = registerIcon('inline-chat', Codicon.sparkle, localize('startInlineChatIcon', 'Icon which spawns the inline chat from the gutter'));
+
+export class InlineChatDecorationsContribution extends Disposable implements IEditorContribution {
+
+	private localToDispose = new DisposableStore();
+	private inlineChatLineNumber: number | undefined;
+	private gutterDecorationID: string | undefined;
+	private gutterDecorationLine: number | undefined;
+
+	public static readonly gutterSettingID = 'inlineChat.showGutterIcon';
+	private static readonly gutterIconClassName = 'codicon-inline-chat';
+
+	private static readonly GUTTER_DECORATION = ModelDecorationOptions.register({
+		description: 'inline-chat-decoration',
+		glyphMarginClassName: ThemeIcon.asClassName(gutterInlineChatIcon),
+		glyphMargin: { position: GlyphMarginLane.Left },
+		stickiness: TrackedRangeStickiness.NeverGrowsWhenTypingAtEdges,
+	});
+
+	constructor(
+		private readonly editor: ICodeEditor,
+		@IInlineChatService private readonly inlineChatService: IInlineChatService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService
+	) {
+		super();
+		this._register(this.configurationService.onDidChangeConfiguration((e: IConfigurationChangeEvent) => {
+			if (!e.affectsConfiguration(InlineChatDecorationsContribution.gutterSettingID)) {
+				return;
+			}
+			this.onEnablementOrModelChanged();
+		}));
+		this._register(this.inlineChatService.onDidChangeProviders(() => this.onEnablementOrModelChanged()));
+		this._register(this.editor.onDidChangeModel(() => this.onEnablementOrModelChanged()));
+	}
+
+	private onEnablementOrModelChanged(): void {
+		// cancels the scheduler, removes editor listeners / removes decoration
+		this.localToDispose.clear();
+		const model = this.editor.getModel();
+		if (!model || !this.isSettingEnabled() || !this.hasProvider()) {
+			return;
+		}
+		const decorationUpdateScheduler = new RunOnceScheduler(() => this.onSelectionOrContentChanged(model), 200);
+		this.localToDispose.add(decorationUpdateScheduler);
+		this.localToDispose.add(this.editor.onDidChangeCursorSelection(() => decorationUpdateScheduler.schedule()));
+		this.localToDispose.add(this.editor.onDidChangeModelContent(() => decorationUpdateScheduler.schedule()));
+		this.localToDispose.add(this.editor.onMouseDown(async (e: IEditorMouseEvent) => {
+			if (!e.target.element?.classList.contains(InlineChatDecorationsContribution.gutterIconClassName) || !this.editor.hasModel()) {
+				return;
+			}
+			const startLineNumber = this.editor.getSelection().startLineNumber;
```

Comment:
If we don't want that the dcoration shows why the inline chat is open, we probably should not draw the decoration in the first place. That should be done in onSelectionOrContentChanged

Reply:
I added it as in that case `this.editor.getSelection();` is always non null. I'll do a regular selection non null check

Reply:
That code has for the purpose to stop the flickering of the inline chat when you click on the icon when the inline chat is already opened, not for removing the decoration.